### ISSUE TITLE
CTC TimeOrder argument inconsistency

### ISSUE THREAD

**Issue by nmiculinic** [tensorflow/tensorflow#6388]
For example, here's signature for dynamicRNN:
```
tf.nn.dynamic_rnn(cell, inputs, sequence_length=None, initial_state=None, dtype=None, parallel_iterations=None, swap_memory=False, time_major=False, scope=None)
```

Here is for ctc_loss:
```
tf.nn.ctc_loss(labels, inputs, sequence_length, preprocess_collapse_repeated=False, ctc_merge_repeated=True, time_major=True)

```

There is the difference between default behavior for **time_major**? (Rest of CTC module is similar in behaving time first.) This is unintuitive and should be changed to batch first in whole CTC module by default to be in consistency with rest of tensorflow library. (Or make other function time first, as long as there's consensus). Conv1D similarly puts batch dimension first, than time one. (Also did other of labels and inputs arguments have changed in ctc_loss between 0.11 and 0.12?)

In enforcing consistency developers using this library have less on their mental workload and enhancing productivity.

**Comment by aselle** [comment]
@ebrevdo, could you comment on this issue?

**Comment by ebrevdo** [comment]
Time major is more efficient with rnns, but dynamic_rnn was made batch major for consistency with the rest of tf.  Since most users of CTC are experts, and are clearly going to be combining it with rnns, we are encouraging users to use time major here as well.  If you're using dynamic_rnn, I encourage you to use the flag time_major=True for efficiency.

**Comment by ebrevdo** [comment]
The order of arguments may have changed. Andrew, were you the one who made that change?  See Martin wickes recent announcement on the tf discussion mailing list.

**Comment by nmiculinic** [comment]
Thanks for the tip, I'll use time major in RNN as well.
